week2/homework/src/index.js

```diff
@@ -1,3 +1,117 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const fs = require('fs');
+const { promisify } = require('util');
+
+const filePath = './list.json';
+const defaultEncoding = 'utf-8';
+
+const writeFilePromisify = promisify(fs.writeFile);
+const appendFilePromisify = promisify(fs.appendFile);
+const readFilePromisify = promisify(fs.readFile);
+
+let todos;
+
+const getAllTodos = async () => {
+  try {
+    const data = await readFilePromisify(filePath, defaultEncoding);
+    todos = JSON.parse(data).todos;
+    return todos;
+  } catch (err) {
+    createEmptyJSONFile();
+  }
+};
+
+const createEmptyJSONFile = async () => {
+  todos = [];
+  const writeData = JSON.stringify({ todos }, null, 2);
+  await writeFilePromisify(filePath, writeData)
+    .then(console.log)
+    .catch(err => console.error(err));
+};
+
+const saveTodos = async () => {
+  try {
+    await writeFilePromisify(filePath, JSON.stringify({ todos }, null, 2));
+  } catch (err) {
+    console.log(err.message);
+  }
+};
+
+const addTodo = async (id, title) => {
+  await getAllTodos();
+  todos.push({ id: id, title: title });
+  await saveTodos();
+};
+
+const updateTodo = async (id, newTitle) => {
+  await getAllTodos();
+  const updatedTodos = todos.map(todo => {
+    if (todo.id === id) {
+      return {
+        id,
+        title: newTitle,
+      };
+    }
+    return todo;
+  });
```

Comment:
Your code is correct but I think the following is simpler and easier to understand

```javascript
let toUpdate = todos.find(todo => todo.id === ud)
toUpdate.title = newTitle;
```

Reply:
hey @gajduk . i tried way of you but I had a problem. toUpdate(it is the new todo) shows only it self. but my updateTodos shows all todos including new todo. how can I handle it?